content/gocourse-2019-05-melbourne.slide

```diff
@@ -840,32 +840,33 @@ Embedding types introduces the possiblity for naming conflicts. Go follows some
 - Add locking for proper use of sync.Map
 - Bonus points: Add a third `Storer` implementation using [[https://github.com/syndtr/goleveldb][LevelDB]]
 
+* Background: Process, Threads and OS Scheduling
 
-* Concurrency (1/2)
+.image _img/process.png
+.caption [[https://randu.org/tutorials/threads/]]
 
-"Don't communicate by sharing memory, share memory by communicating."
-"Concurrency is not parallelism."
-"Concurrency is about structure, parallelism is about execution."
- - [[https://go-proverbs.github.io/][Rob Pike]] ([[https://www.youtube.com/watch?v=cN_DpYBzKso][Talk]], [[https://talks.golang.org/2012/waza.slide][Slides]])
+- Large (default 2 Mb on Linux) stack size
+- 1x1 mapping between user and kernel threads
 
-*Overview:* *built-in* *language* *features*
-Goroutines (`go`)
-Channels (`chan`, `<-`)
-Select (`select`)
+* Go Runtime
 
+- Scheduling, Memory Management, Garbage Collection
 
-* Concurrency (2/2)
+.image _img/go_runtime.png
+.caption [[http://bit.ly/30uMfhm]]
 
-Concurrency isn't easy - we are only scratching the surface.
+* Goroutines and Scheduler
 
-- [[https://goplay.space/#YtuJgIuIswH][Goroutines]] ([[https://play.golang.org/p/YtuJgIuIswH][Playground]]`*`)
-- [[https://goplay.space/#5M0f3-YY44W][Channels]] ([[https://play.golang.org/p/5M0f3-YY44W][Playground]]`*`)
-- [[https://goplay.space/#FEWHmUA0QtN][Buffered Channels]]
-- [[https://goplay.space/#ANO8rpfNynY][Range and close]] ([[https://play.golang.org/p/ANO8rpfNynY][Playground]]`*`)
-- [[https://goplay.space/#R1qgjOsPSGk][Select]] ([[https://play.golang.org/p/R1qgjOsPSGk][Playground]]`*`)
-- [[https://goplay.space/#4TGM29YbvX4][Deadlocks]] ([[https://play.golang.org/p/4TGM29YbvX4][Playground]]`*`)
+- Goroutine is an extremely lightweight thread of execution
+- Low initial stack size (2 Kb) that grows
+- Runs concurrently like threads
+- Scheduler tracks goroutines and schedules them to run on a set of OS threads
+- GOMAXPROCS and runtime.GOMAXPROCS() sets the number of OS threads
+- Start a goroutine [[https://play.golang.org/p/2sr6XrkGYKz][using a `go` statement]]
```

Comment:
This looks a little strange on the slides with the bold font.
recommendation: `[using a 'go' statement]`

Reply:
`[using the 'go' statement]`

Reply:
Done